pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -1617,47 +1624,80 @@ func TestPermitWaitingMetric(t *testing.T) {
 			}
 
 			f.RunPermitPlugins(context.TODO(), nil, pod, "")
+			f.WaitOnPermit(context.TODO(), pod)
 
 			collectAndComparePermitWaitDuration(t, tt.wantRes)
 		})
 	}
 }
 
-func TestRejectWaitingPod(t *testing.T) {
+func TestWaitOnPermit(t *testing.T) {
 	pod := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "pod",
 			UID:  types.UID("pod"),
 		},
 	}
 
-	testPermitPlugin := &TestPermitPlugin{}
-	r := make(Registry)
-	r.Register(permitPlugin,
-		func(_ *runtime.Unknown, fh FrameworkHandle) (Plugin, error) {
-			return testPermitPlugin, nil
-		})
-	plugins := &config.Plugins{
-		Permit: &config.PluginSet{Enabled: []config.Plugin{{Name: permitPlugin, Weight: 1}}},
+	tests := []struct {
+		name        string
+		action      func(f Framework)
+		wantStatus  Code
+		wantMessage string
+	}{
+		{
+			name: "Reject Waiting Pod",
+			action: func(f Framework) {
+				f.GetWaitingPod(pod.UID).Reject("reject message")
+			},
+			wantStatus:  Unschedulable,
+			wantMessage: "pod \"pod\" rejected while waiting on permit: reject message",
+		},
+		{
+			name: "Allow Waiting Pod",
+			action: func(f Framework) {
+				f.GetWaitingPod(pod.UID).Allow(permitPlugin)
+			},
+			wantStatus:  Success,
+			wantMessage: "",
+		},
 	}
 
-	f, err := newFrameworkWithQueueSortAndBind(r, plugins, emptyArgs)
-	if err != nil {
-		t.Fatalf("Failed to create framework for testing: %v", err)
-	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testPermitPlugin := &TestPermitPlugin{}
+			r := make(Registry)
+			r.Register(permitPlugin,
+				func(_ *runtime.Unknown, fh FrameworkHandle) (Plugin, error) {
+					return testPermitPlugin, nil
+				})
+			plugins := &config.Plugins{
+				Permit: &config.PluginSet{Enabled: []config.Plugin{{Name: permitPlugin, Weight: 1}}},
+			}
 
-	go func() {
-		for {
-			waitingPod := f.GetWaitingPod(pod.UID)
-			if waitingPod != nil {
```

Comment:
Note that this type of checking is no longer necessary after moving `RunPermitPlugins` to the scheduling cycle.